File: datalib/src/main/java/nl/acidcats/tumblrlikes/core/usecases/photos/PhotoFilterUseCase.java
package nl.acidcats.tumblrlikes.core.usecases.photos;

import nl.acidcats.tumblrlikes.core.constants.FilterType;
import rx.Observable;

/**
 * Created on 19/10/2018.
 */
public interface PhotoFilterUseCase {
    Observable<Boolean> storeFilterSelection(FilterType filterType);

    Observable<FilterType> getSelectedFilterType();
}

File: datalib/src/main/kotlin/nl/acidcats/tumblrlikes/core/usecases/photos/PhotoFilterUseCase.kt
package nl.acidcats.tumblrlikes.core.usecases.photos

import nl.acidcats.tumblrlikes.core.constants.FilterType
import rx.Observable

/**
 * Created on 19/10/2018.
 */
interface PhotoFilterUseCase {
    fun getSelectedFilterType(): Observable<FilterType>
    fun storeFilterSelection(filterType: FilterType): Observable<FilterType>
}
